### PR TITLE
Lcam 820

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-820)

Describe what you did and why.

## Checklist

Align the Prosecution Concluded SQS listener with Maat API

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crown-court-proceeding/build.gradle [hunk 1]
```diff
@@ -46,6 +49,12 @@ dependencies {
     implementation "io.sentry:sentry-logback"
     implementation "io.sentry:sentry-spring-boot-starter"
 
+    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-aws', version: '2.2.6.RELEASE'
+    implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-aws-messaging', version: '2.2.6.RELEASE'
```
Comment (venkat980) [inline crown-court-proceeding/build.gradle:53]: Should we define a constant for the version and use it here?

File: crown-court-proceeding/build.gradle [hunk 2]
```diff
@@ -7,6 +7,7 @@ plugins {
     id "org.sonarqube" version "4.0.0.2929"
     id "info.solidsoft.pitest" version "1.9.11"
     id "org.jsonschema2dataclass" version "6.0.0"
+    id 'io.spring.dependency-management' version '1.1.0'
```
Comment (skasthuri36326) [inline crown-court-proceeding/build.gradle:10]: Why we need io.spring.dependency-management plugin when we have the spring boot plugin?

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/ProsecutionConcludedListenerTest.java [hunk 3]
```diff
@@ -72,6 +77,16 @@ void givenJSONMessageIsReceived_whenProsecutionConcludedListenerIsInvoked_thenRe
         softly.assertAll();
     }
```
Comment (skasthuri36326) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/ProsecutionConcludedListenerTest.java:79]: localstack provides SQS test queues for integration test. Example can be found in [laa-maat-court-data-api](https://github.com/ministryofjustice/laa-maat-court-data-api/blob/main/maat-court-data-api/src/test/java/gov/uk/courtdata/link/service/CreateLinkListenerEndToEndTest.java)

File: crown-court-proceeding/build.gradle (hunk 2 above)
Comment (muthusubra) [reply crown-court-proceeding/build.gradle:10]: Used internally by Spring FW cloud, so it is required. Created a separate story to potentially fully migrate to Spring Cloud V3

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/ProsecutionConcludedListenerTest.java (hunk 3 above)
Comment (muthusubra) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/ProsecutionConcludedListenerTest.java:79]: Created a separate story to potentially implement local stack for testing

File: crown-court-proceeding/build.gradle (hunk 1 above)
Comment (muthusubra) [reply crown-court-proceeding/build.gradle]: Updated

File: crown-court-proceeding/build.gradle [hunk 4]
```diff
@@ -46,6 +50,12 @@ dependencies {
     implementation "io.sentry:sentry-logback"
     implementation "io.sentry:sentry-spring-boot-starter"
 
+    implementation "org.springframework.cloud:spring-cloud-starter-aws:$versions.springFrameworkCloudVersion"
+    implementation "org.springframework.cloud:spring-cloud-starter-aws-messaging:$versions.springFrameworkCloudVersion"
+    implementation "org.springframework.cloud:spring-cloud-aws-autoconfigure:$versions.springFrameworkCloudVersion"
```
Comment (skasthuri36326) [inline crown-court-proceeding/build.gradle:55]: As per the [release notes](https://github.com/awspring/spring-cloud-aws): Spring Cloud AWS 3.0.x is compatible with Spring Boot 3.0.x. As per the [Spring Guide](https://docs.awspring.io/spring-cloud-aws/docs/3.0.0/reference/html/index.html#sqs-integration) we can use the spring-cloud-aws-starter-sqs

Comment (muthusubra) [reply crown-court-proceeding/build.gradle:55]: Sri, As you can see, that is what I've used and tested and it is working fine. Not sure if I'm missing anything here

I thought I clarified everything in our call. I  don't see a point in delaying this as this is our last day of Sprint

If you till have doubts, I'm happy to get on a call and discuss to close this. Back and forth comments is not going to help

Comment (muthusubra) [reply crown-court-proceeding/build.gradle:55]: As discussed, this will be addressed as part of
https://dsdmoj.atlassian.net/browse/LCAM-833

**Pull request merged**